File: Practica3/Marco teorico/2. (SRP)/SRP.java
import java.math.*;

class Empleado {
    private String nombre;
    private int edad;
    private String direccion;

    public Empleado(String nombre, int edad, String direccion) {
        this.nombre = nombre;
        this.edad = edad;
        this.direccion = direccion;
    }

    public void mostrarInformacionPersonal() {
        System.out.println("Nombre: " + this.nombre);
        System.out.println("Edad: " + this.edad);
        System.out.println("Dirección: " + this.direccion);
    }
}

class Salario {
    private double salarioBase;
    private double bonificaciones;

    public Salario(double salarioBase, double bonificaciones) {
        this.salarioBase = salarioBase;
        this.bonificaciones = bonificaciones;
    }

    public double calcularSalarioTotal() {
        return this.salarioBase + this.bonificaciones;
    }
}

class Proyecto {
    private String nombreProyecto;
    private String descripcion;

    public Proyecto(String nombreProyecto, String descripcion) {
        this.nombreProyecto = nombreProyecto;
        this.descripcion = descripcion;
    }

    public void mostrarInformacionProyecto() {
        System.out.println("Nombre del proyecto: " + this.nombreProyecto);
        System.out.println("Descripción: " + this.descripcion);
    }
}

public class Main {
    public static void main(String[] args) {
        Empleado emp1 = new Empleado("Juan Perez", 25, "Av. Puente Chilina 123");
        emp1.mostrarInformacionPersonal();

        Salario sal1 = new Salario(1000, 200);
        System.out.println("Salario total: $" + sal1.calcularSalarioTotal());

        ArrayList<Proyecto> proyectos = new ArrayList<>();
        proyectos.add(new Proyecto("Proyecto 1", "Desarrollo de software"));
        proyectos.add(new Proyecto("Proyecto 2", "Implementación de redes"));

        for (Proyecto p : proyectos) {
            p.mostrarInformacionProyecto();
        }
    }
}

File: Practica3/Marco teorico/2. (SRP)/SRP.py
import math
class Empleado:
    def __init__(self, nombre, edad, direccion):
        self.nombre = nombre
        self.edad = edad
        self.direccion = direccion

    def mostrarInformacionPersonal(self):
        print(f"Nombre: {self.nombre}")
        print(f"Edad: {self.edad}")
        print(f"Dirección: {self.direccion}")

class Salario:
    def __init__(self, salarioBase, bonificaciones):
        self.salarioBase = salarioBase
        self.bonificaciones = bonificaciones

    def calcularSalarioTotal(self):
        return self.salarioBase + self.bonificaciones

class Proyecto:
    def __init__(self, nombreProyecto, descripcion):
        self.nombreProyecto = nombreProyecto
        self.descripcion = descripcion

    def mostrarInformacionProyecto(self):
        print(f"Nombre del proyecto: {self.nombreProyecto}")
        print(f"Descripción: {self.descripcion}")


emp1 = Empleado("Juan Perez", 25, "Av. Puente Chilina 123")
emp1.mostrarInformacionPersonal()

sal1 = Salario(1000, 200)
print(f"Salario total: ${sal1.calcularSalarioTotal()}")

proyectos = []
proyectos.append(Proyecto("Proyecto 1", "Desarrollo de software"))
proyectos.append(Proyecto("Proyecto 2", "Implementación de redes"))

for p in proyectos:
    p.mostrarInformacionProyecto()

